### DISCUSSION TITLE
c#, mouse on background WebView2

### DISCUSSION THREAD

**Discussion by Rankors** [MicrosoftEdge/WebView2Feedback#4312] [Q&A]
Hi all. I have a question, I’m making a program (C#) for college, this program is essentially an analogue of a wallpaper engine, I already understand how to set a window to the background. But there is an error, there is WebView2 in the window, and it does not accept data from the mouse, tell me, is it possible to solve this somehow?

**Comment by DanielSparta** [comment]
Tell me what your trying to do and I will trying help you, Please give move information about what you trying to do

**Reply by Rankors** [reply]
I did it like in this project -> https://www.codeproject.com/Articles/856020/Draw-Behind-Desktop-Icons-in-Windows-plus .  In the Form2 , I placed the WebView2 element, but Webview2 and the site do not respond to the mouse

**Reply by DanielSparta** [reply]
But what are you trying to do? Also, what is the Form2? I dont see any Form2. please provide me the code your using right now.
Tell me more details about what your trying to do please, And provide more information.

**Reply by Rankors** [reply]
Form2 have element WebView2, and im run any site, and my dll set Form2 behind desktop icons `#include <Windows.h>
#include "pch.h"
#include <iostream>
#define _UNICODE
HWND wallpaper = NULL;
bool set_wallpapers(HWND hwnd);

extern "C" __declspec(dllexport) bool set(HWND hwnd)
{
	if (hwnd)
	{
		if (set_wallpapers(hwnd))
		{
			return true;
		}
	}
	return false;
}

HWND get_wallpaper() {
	HWND progman = FindWindow(L"Progman", NULL);
	SendMessageTimeout(progman, 0x052C, 0, 0, 0x0000, 1000, nullptr);
	EnumWindows([](HWND hwnd, LPARAM lparam) -> BOOL CALLBACK
		{
			HWND p = FindWindowEx(hwnd, NULL, L"SHELLDLL_DefView", NULL);
	if (p)
	{
		wallpaper = FindWindowEx(NULL, hwnd, L"WorkerW", NULL);
	}
	return true;
		}, NULL
	);
	return wallpaper;
}

bool set_wallpapers(HWND hwnd)
{
	HWND desktop = get_wallpaper();
	if (desktop)
	{
		SetParent(hwnd, desktop);
		if (SetParent(hwnd, desktop))
		{
			return true;
		}
	}
	return false;
}

BOOL APIENTRY DllMain( HMODULE hModule,
                       DWORD  ul_reason_for_call,
                       LPVOID lpReserved
                     )
{
    switch (ul_reason_for_call)
    {
    case DLL_PROCESS_ATTACH:
    case DLL_THREAD_ATTACH:
    case DLL_THREAD_DETACH:
    case DLL_PROCESS_DETACH:
        break;
    }
    return TRUE;
}

` its dll code `public WebWallpaper()
        {
            InitializeComponent();} 
            ` its Form2 code

**Reply by DanielSparta** [reply]
Ok, And what exactly your trying to do?

**Reply by Rankors** [reply]
I want to install Form2 behind the windows desktop icons, i.e. set instead of desktop background

**Reply by DanielSparta** [reply]
Your trying to put a new desktop background picture, using C#?

**Reply by Rankors** [reply]
Form2 send him HWND, to dll(c++), in dll code im put to Form2 behind icons

**Reply by DanielSparta** [reply]
But what your trying to do? to put a new desktop background picture?

**Reply by Rankors** [reply]
In short, I open the browser. Next, I move the browser window behind the desktop icons
here's a photo of how it's done. i set the site to the background instead of standard images.
here is the site that I set as my desktop wallpaper https://mattatz.github.io/GPUFluidWireframe/
![изображение_2024-01-30_210533461](https://github.com/MicrosoftEdge/WebView2Feedback/assets/94163101/23d7364a-fff1-4961-b623-7dea0c3a9f6c)

**Reply by DanielSparta** [reply]
And when you tried that on a simple winform, that work?

**Reply by Rankors** [reply]
da
